libs/libc/pthread/pthread_keycreate.c

```diff
@@ -70,23 +72,45 @@
 int pthread_key_create(FAR pthread_key_t *key,
                        CODE void (*destructor)(FAR void *))
 {
-  int tlsindex;
+  FAR struct task_info_s *info = task_get_info();
+  int candidate;
+  int ret;
 
   DEBUGASSERT(key != NULL);
+  DEBUGASSERT(info != NULL);
 
-  /* Allocate a TLS index */
+  /* Search for an unused index.  This is done in a critical section here to
+   * avoid concurrent modification of the group TLS index set.
+   */
 
-  tlsindex = tls_alloc(destructor);
+  ret = _SEM_WAIT(&info->ta_sem);
 
-  /* Check if found a TLS index. */
+  if (ret < 0)
+    {
+      ret = _SEM_ERRNO(ret);
+      return ret;
+    }
```

Comment:
I think we need to wait for https://github.com/apache/incubator-nuttx/pull/6320 to be merged first as according to POSIX: `The pthread_key_create() function shall not return an error code of [EINTR]`

Reply:
Do you want to replace sem with nxmutex? it's better to wait @anjiahao1 do the replacement in the batch.

Reply:
@anjiahao1 is preparing the patch to replace sem lock with mutex, so let's wait his patch instead.